src/nvim/eval/typval_encode.h

```diff
@@ -248,6 +253,10 @@ static int name##_convert_one_value(firstargtype firstargname, \
       TYPVAL_ENCODE_CONV_FUNC(tv->vval.v_string); \
       break; \
     } \
+    case VAR_PARTIAL: { \
+      TYPVAL_ENCODE_CONV_PARTIAL(tv->vval.v_partial); \
```

Comment:
This function needs to do it's work iterative, but here it is possible to call `string_convert_one_value()` recursively, if the dictionary of the partial contains the partial:

```vim
func Callback() 
endf

let myDict={}
let Cb = function('Callback', [], myDict)
let myDict["1"]=Cb
echo Cb
```
```
=================================================================
==17185==ERROR: AddressSanitizer: stack-overflow on address 0x7fffbe461e5c (pc 0x7f17ad3888ae bp 0x7fffbe462300 sp 0x7fffbe461d80 T0)
    #0 0x7f17ad3888ad in _IO_vfprintf (/lib64/libc.so.6+0x468ad)
    #1 0x7f17ad38b660  (/lib64/libc.so.6+0x49660)
    #2 0x7f17ad388a54 in _IO_vfprintf (/lib64/libc.so.6+0x46a54)
    #3 0x4c65bf in __interceptor_vfprintf /home/oni-link/git/llvm/projects/compiler-rt/lib/asan/../sanitizer_common/sanitizer_common_interceptors.inc:1342
    #4 0x4c6692 in __interceptor_fprintf /home/oni-link/git/llvm/projects/compiler-rt/lib/asan/../sanitizer_common/sanitizer_common_interceptors.inc:1387
    #5 0x9b83d0 in string_convert_one_value /home/oni-link/git/neovim/src/nvim/eval/encode.c:443:1
    #6 0x996fcb in encode_vim_to_string /home/oni-link/git/neovim/src/nvim/eval/encode.c:443:1
    #7 0x991e4a in encode_tv2string /home/oni-link/git/neovim/src/nvim/eval/encode.c:811:3
    #8 0x9b8a54 in string_convert_one_value /home/oni-link/git/neovim/src/nvim/eval/encode.c:443:1
    #9 0x996fcb in encode_vim_to_string /home/oni-link/git/neovim/src/nvim/eval/encode.c:443:1
    #10 0x991e4a in encode_tv2string /home/oni-link/git/neovim/src/nvim/eval/encode.c:811:3
    #11 0x9b8a54 in string_convert_one_value /home/oni-link/git/neovim/src/nvim/eval/encode.c:443:1
    #12 0x996fcb in encode_vim_to_string /home/oni-link/git/neovim/src/nvim/eval/encode.c:443:1
    #13 0x991e4a in encode_tv2string /home/oni-link/git/neovim/src/nvim/eval/encode.c:811:3
    #14 0x9b8a54 in string_convert_one_value /home/oni-link/git/neovim/src/nvim/eval/encode.c:443:1
    .
    .
    .
```
So `TYPVAL_ENCODE_CONV_PARTIAL` should not make recursive calls to `string_convert_one_value()` by calling `encode_tv2string()`, instead it should push work on the stack like in the case for `VAR_DICT`.